### PR TITLE
fix(core): Fix several misspellings 

### PR DESCRIPTION
**Pull request by b-long**
This PR fixes several misspellings.

Since doing this work manually seems like a waste of time, you might consider introducing automation to prevent spelling errors (I make them constantly).
 
One automatic spell-checker is https://github.com/codespell-project/codespell .  It's very easy to run both locally and in CI, using pre-commit, e.g. the following will lint Go, Markdown, and YAML:

```yaml
# To learn about "pre-commit", see:
#   https://github.com/pre-commit/pre-commit

repos:
-   repo: https://github.com/codespell-project/codespell
    rev: v2.2.6
    hooks:
    - id: codespell
      files: ^.*\.(go|md|yaml)$
      args: ["--write-changes", "--ignore-words-list", "MYCOOLWORD,ANOTHAONE"]
```

### CODE REVIEW

**Comment by jrschumacher** [comment]
This is a great idea! We already use golangci-lint which has the ability to add [misspell](https://golangci-lint.run/usage/linters/#misspell) to check for spelling errors. I'd prefer to minimize the number of external tools if we can. Would you give this a try to see how it compares?

Also I'm not sure we want to introduce precommits as this will need to be a more detailed conversation. Precommits are great for small teams but large opensource projects they can become burdensome and many projects lean on CI linters.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# To learn about "pre-commit", see:
+#   https://github.com/pre-commit/pre-commit
+repos:
+-   repo: https://github.com/codespell-project/codespell
+    rev: v2.2.6
+    hooks:
+    - id: codespell
+      files: ^.*\.(go|md|yml|yaml|sh)$
+      args: ["--write-changes", "--ignore-words-list", "MYCOOLWORD,ANOTHAONE"]
```
Comment (dmihalcik-virtru) [inline .pre-commit-config.yaml:9]: I'm guessing
```suggestion
      args: ["--write-changes", "--ignore-words-list", "virtru,opentdf"]
```

File: .github/workflows/lint-all.yaml [hunk 2]
```diff
@@ -44,3 +44,11 @@ jobs:
       - uses: bufbuild/buf-lint-action@044d13acb1f155179c606aaa2e53aea304d22058
         with:
           input: service
+  
+  pre-commit:
```
Comment (dmihalcik-virtru) [inline .github/workflows/lint-all.yaml:48]: Can we move this to checks.yaml and delete lint-all?

**Review by jrschumacher — APPROVED** [review]

File: .github/workflows/lint-all.yaml (hunk 2 above)
Comment (b-long) [reply .github/workflows/lint-all.yaml]: Resolved by deleting this bit of code 👍

File: .pre-commit-config.yaml (hunk 1 above)
Comment (b-long) [reply .pre-commit-config.yaml]: Resolved by deleting this bit of code 👍

**Pull request merged**